File: api/src/main/java/com/ricky/adocao/enums/PetGeneroEnum.java
package com.ricky.adocao.enums;

public enum PetGeneroEnum {
    NONE,
    MACHO,
    FEMEA;

    public String getDecricao() {
        return switch (this) {
            case NONE -> "";
            case MACHO -> "PET_GENERO_MACHO";
            case FEMEA -> "PET_GENERO_FEMEA";
            default -> throw new IllegalArgumentException("Unexpected value: " + this);
        };
    }

    @Override
    public String toString() {
        return getDecricao().replace("PET_GENERO_", "");
    }
}

File: app/app/src/main/java/com/ricky/adocaoapp/domain/enums/PetGeneroEnum.kt
package com.ricky.adocaoapp.domain.enums

enum class PetGeneroEnum {
    MACHO,
    FEMEA;

    override fun toString(): String {
        return when(this){
            MACHO -> "Macho"
            FEMEA -> "Fêmea"
        }
    }


}